### PR TITLE
docs - CREATE FUNCTION - new attribute EXECUTE ON INITPLAN

### PR DESCRIPTION
**Pull request by mkiyama**
Updated
CREATE FUNCTION
ALTER FUNCTION
Using Functions and Operators

This will be backported to 6X_STABLE

HTML output on temporary GPDB doc review site
https://docs-msk-gpdb6-dev.cfapps.io/7-0/ref_guide/sql_commands/CREATE_FUNCTION.html

https://docs-msk-gpdb6-dev.cfapps.io/7-0/ref_guide/sql_commands/ALTER_FUNCTION.html
https://docs-msk-gpdb6-dev.cfapps.io/7-0/admin_guide/query/topics/functions-operators.html

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/query/topics/functions-operators.xml [hunk 1]
```diff
@@ -126,6 +126,13 @@
                                 master.</entry>
                             <entry colname="col4"/>
                         </row>
+                        <row>
+                            <entry colname="col1">EXECUTE ON INITPLAN</entry>
+                            <entry colname="col3">Indicates that the function is executed on segment
+                                instances by a Query Executor as part of an InitPlan node when
+                                possible. </entry>
```
Comment (zhangh43) [inline gpdb-doc/dita/admin_guide/query/topics/functions-operators.xml]: `segment instances by a Query Executor` not correct.
It's a little complicated.
`EXECUTE ON INITPLAN`indicates the function will run twice. The first run will execute the function and is run as an InitPlan on Master instance and store the result into in a temp storage(tuplestore). InitPlan runs ahead of the MainPlan. Then the second run is in the MainPlan where the function is run as EntryDB and will not execute the function body, instead just fetching results from temp storage directly. (EntryDB is a special Query Executor which run on Master instance).
`when possible` is needed. Please also note that this is just an experimental feature, whose use case islimited to `create table t as select * from udf()` and `insert into t  select * from udf()` this kind of queries currently.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 2]
```diff
@@ -434,7 +451,68 @@ $SomeTag$Dianne's horse$SomeTag$</codeblock>
...
+                    function did not contain the <codeph>EXECUTE ON INITPLAN</codeph> attribute, the
+                    CTAS command returns the error <codeph>function cannot execute on a QE
+                        slice</codeph>.</p><p>When a function uses the <codeph>EXECUTE ON
+                        INITPLAN</codeph> attribute, a command that uses the function such as
+                        <codeph>CREATE TABLE t SELECT * FROM get_data()</codeph> gathers the results
+                    of the function onto the master segment and then redistributes the results to
+                    segment instances when inserting the data. If the function returns a large
+                    amount of data, the master might become a bottleneck when gathering and
+                    redistributing data. Performance might improve if you rewrite the function to
+                    run the CTAS command in the user defined function and use the table name as an
```
Comment (zhangh43) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: Like this best practice. 
```
indicates that the function is executed on segment instances by a Query Executor as part of an InitPlan node when possible.
```
The above explanation happens several times, they all need to be replace by the 'twice run' comments.

File: gpdb-doc/dita/admin_guide/query/topics/functions-operators.xml (hunk 1 above)
Comment (mkiyama) [reply gpdb-doc/dita/admin_guide/query/topics/functions-operators.xml]: Updated description of EXECUTE ON INITPLAN based on review comments.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 3]
```diff
@@ -434,7 +461,72 @@ $SomeTag$Dianne's horse$SomeTag$</codeblock>
...
+                    you view the query plan of the CTAS command with <codeph>EXPLAIN ANALYZE
+                        VERBOSE</codeph>, the plan shows that the function is run as part of an
+                    InitPlan node and one of the listed slices is labeled as <codeph>entry
+                        db</codeph>. The query plan of a simple CTAS command without the function
+                    does not have an InitPlan node or an <codeph>entry db</codeph> slice.</p><p>If
+                    the function did not contain the <codeph>EXECUTE ON INITPLAN</codeph> attribute,
+                    the CTAS command returns the error <codeph>function cannot execute on a QE
+                        slice</codeph>.</p><p>When a function uses the <codeph>EXECUTE ON
+                        INITPLAN</codeph> attribute, a command that uses the function such as
+                        <codeph>CREATE TABLE t SELECT * FROM get_data()</codeph> gathers the results
```
Comment (zhangh43) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: `CREATE TABLE t SELECT * FROM get_data()` to `CREATE TABLE t AS SELECT * FROM get_data()`

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 4]
```diff
@@ -434,7 +461,72 @@ $SomeTag$Dianne's horse$SomeTag$</codeblock>
                             <codeph>FROM</codeph> clause.</li>
                     <li>A query that includes the function falls back from GPORCA to the Postgres
                         Planner.</li>
-                </ul></sectiondiv></section>
+                </ul><p>The Beta attribute <codeph>EXECUTE ON INITPLAN</codeph> indicates that the
```
Comment (zhangh43) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: Please remove the Beta.
This explanation looks good to me. Thanks @mkiyama

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 5]
```diff
@@ -268,10 +273,32 @@ SELECT foo();</codeblock><p>In
...
                         invocation. The overall result of the function is the <codeph>UNION
                             ALL</codeph> of the results from all segment instances. </pd>
+                    <pd><codeph>EXECUTE ON INITPLAN</codeph> indicates that the function contains an
+                        SQL command that dispatches queries to the segment instances and requires
+                        special processing on the master instance by Greenplum Database when
+                        possible.
+                        <note><codeph>EXECUTE ON INITPLAN</codeph> is a Beta feature <ph
+                                otherprops="pivotal">and is not supported in a production
+                                environment</ph>. The attribute is only supported in functions that
+                            are used in the <codeph>FROM</codeph> clause of a of a <codeph>CREATE
```
Comment (zhangh43) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: Remove Beta notice, but keep the comments which explains when to use it:`The attribute is only supported in functions that are used in ...`

File: gpdb-doc/dita/ref_guide/sql_commands/ALTER_FUNCTION.xml [hunk 6]
```diff
@@ -120,11 +121,15 @@ RESET ALL</codeblock></section>
             all primary segment instances, but not the master, for each invocation. The overall
             result of the function is the <codeph>UNION ALL</codeph> of the results from all segment
             instances. </pd>
+          <pd><codeph>EXECUTE ON INITPLAN</codeph> indicates that the function contains an SQL
```
Comment (zhangh43) [inline gpdb-doc/dita/ref_guide/sql_commands/ALTER_FUNCTION.xml]: an to a

**Pull request merged**